src/lib.rs

```diff
@@ -115,15 +120,64 @@ fn print_notice(name: &str, current_version: &str, latest_version: &str) {
     print!("{}", generate_notice(name, current_version, latest_version));
 }
 
-/// Validates current version of crate
-/// Takes the current name and version
-/// Prints directly to stdout (Will probably change to be more asynchrounos)
-pub fn check_version(name: &str, current_version: &str) -> Result<(), Box<dyn std::error::Error>> {
+#[derive(Deserialize, Serialize)]
+struct Config {
+    last_checked: DateTime<Utc>,
+}
+
+fn get_app_name(name: &str) -> String {
+    let mut app_name: String = String::from(name);
+    app_name.push_str("-update-notifier");
+    #[cfg(test)]
+    app_name.push_str("-test");
```

Comment:
an idea for `Configstore`, we could potentially allow a `Memory` only store that doesn't write to disk, would be useful for writing tests